examples/blaze/src/main/scala/com/example/http4s/blaze/demo/server/Module.scala

```diff
@@ -0,0 +1,65 @@
+package com.example.http4s.blaze.demo.server
+
+import cats.effect.Effect
+import cats.syntax.semigroupk._ // For <+>
+import com.example.http4s.blaze.demo.server.endpoints._
+import com.example.http4s.blaze.demo.server.endpoints.auth.{BasicAuthHttpEndpoint, GitHubHttpEndpoint}
+import com.example.http4s.blaze.demo.server.service.{FileService, GitHubService}
+import fs2.Scheduler
+import org.http4s.HttpService
+import org.http4s.client.Client
+import org.http4s.server.HttpMiddleware
+import org.http4s.server.middleware.{AutoSlash, ChunkAggregator, GZip, Timeout}
+
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class Module[F[_]](client: Client[F])(implicit F: Effect[F], S: Scheduler) {
+
+  private val fileService = new FileService[F]
+
+  private val gitHubService = new GitHubService[F](client)
+
+  def middleware: HttpMiddleware[F] = {
+    {(service: HttpService[F]) => GZip(service)(F)} compose
+      { service => AutoSlash(service)(F) }
+  }
+
+  val fileHttpEndpoint: HttpService[F] =
+    new FileHttpEndpoint[F](fileService).service
+
+  val nonStreamFileHttpEndpoint = ChunkAggregator(fileHttpEndpoint)
+
+  private val hexNameHttpEndpoint: HttpService[F] =
+    new HexNameHttpEndpoint[F].service
+
+  private val compressedEndpoints: HttpService[F] =
+    middleware(hexNameHttpEndpoint)
+
+  private val timeoutHttpEndpoint: HttpService[F] =
+    new TimeoutHttpEndpoint[F].service
+
+  private val timeoutEndpoints: HttpService[F] =
+    Timeout(1.second)(timeoutHttpEndpoint)
+
+  private val mediaHttpEndpoint: HttpService[F] =
+    new JsonXmlHttpEndpoint[F].service
+
+  private val multipartHttpEndpoint: HttpService[F] =
+    new MultipartHttpEndpoint[F](fileService).service
+
+  private val gitHubHttpEndpoint: HttpService[F] =
+    new GitHubHttpEndpoint[F](gitHubService).service
+
+  val basicAuthHttpEndpoint: HttpService[F] =
+    new BasicAuthHttpEndpoint[F].service
+
+  // NOTE: If you mix services wrapped in `AuthMiddleware[F, ?]` the entire namespace will be protected.
```

Comment:
Note: this is changing with #1665. I'm bringing back the possibility of fallthrough so while it's good to note this, 
there's also the fallthrough options now.

Reply:
Good to know! I guess we can leave it as is and modify when that PR is merged.

Reply:
It's merged now, so you can use `.withFallthrough` IIRC.